src/lib/fetch.ts

```diff
@@ -91,11 +102,42 @@ export async function azleFetch(input: any, init?: any): Promise<any> {
             text: async () => {
                 return contents.toString();
             }
-        };
+        } as any;
     }
 
-    if (url.protocol === 'http:') {
-        // TODO http call to the management canister
+    if (url.protocol === 'https:' || url.protocol === 'http:') {
+        const body = await prepareRequestBody(init);
+        // TODO also do headers and method and everything like on the client?
```

Comment:
Should this be addressed before merging in?

Reply:
Also whatever you do here keep in mind that the icp protocol currently disregards headers and methods

Reply:
I have added support for headers and I have made it so that only GET, HEAD, and POST are supported